packages/webview_flutter/webview_flutter_wkwebview/lib/src/v4/src/webkit_webview_controller.dart

```diff
@@ -17,14 +17,59 @@ import '../../foundation/foundation.dart';
 import '../../web_kit/web_kit.dart';
 import 'webkit_proxy.dart';
 
+/// Media types that can require a user gesture to begin playing.
+///
+/// See [WebKitWebViewControllerCreationParams.mediaTypesRequiringUserAction].
+enum PlaybackMediaTypes {
+  /// No media types require a user gesture to begin playing.
+  none,
+
+  /// A media type that contains audio.
+  audio,
+
+  /// A media type that contains video.
+  video,
+
+  /// All media types require a user gesture to begin playing.
+  all;
+
+  WKAudiovisualMediaType _toWKAudiovisualMediaType() {
+    switch (this) {
+      case PlaybackMediaTypes.none:
+        return WKAudiovisualMediaType.none;
+      case PlaybackMediaTypes.audio:
+        return WKAudiovisualMediaType.audio;
+      case PlaybackMediaTypes.video:
+        return WKAudiovisualMediaType.video;
+      case PlaybackMediaTypes.all:
+        return WKAudiovisualMediaType.all;
+    }
+  }
+}
+
 /// Object specifying creation parameters for a [WebKitWebViewController].
 @immutable
 class WebKitWebViewControllerCreationParams
     extends PlatformWebViewControllerCreationParams {
   /// Constructs a [WebKitWebViewControllerCreationParams].
   WebKitWebViewControllerCreationParams({
     @visibleForTesting this.webKitProxy = const WebKitProxy(),
-  }) : _configuration = webKitProxy.createWebViewConfiguration();
+    Set<PlaybackMediaTypes>? mediaTypesRequiringUserAction,
+  })  : assert(mediaTypesRequiringUserAction == null ||
+            !mediaTypesRequiringUserAction.contains(PlaybackMediaTypes.none) ||
```

Comment:
If we never allow `PlaybackMediaTypes.none`, do we need the `PlaybackMediaTypes.none` enum member?

Reply:
It allows `none`, it just doesn't allow it to exist with another value. I suppose this is confusing though. I removed the `all` and `none` enums and now an empty set will represent none.